### PR TITLE
rgw: remove RGWRegionMap and RGWZoneGroupMap

### PR DESCRIPTION
**Pull request by cbodley**
these are still left over from the radosgw-agent, before multisite v2 in jewel. we kept them around for the conversion logic, but that was only necessary to support upgrades to jewel and kraken

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_rest_config.h [hunk 1]
```diff
@@ -1,88 +0,0 @@
...
-    if (old_format) {
-      return "get_region_map";
-    } else {
-      return "get_zonegroup_map";
-    }
-  }
-};
-
-class RGWOp_ZoneConfig_Get : public RGWRESTOp {
-  RGWZoneParams zone_params;
```
Comment (cbodley) [inline src/rgw/rgw_rest_config.h:47]: oops, the ragweed tests are still using this `/admin/config` interface to read the local zone params

**Review by dang — APPROVED** [review]

**Comment by cbodley** [comment]
jenkins test make check

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
jenkins test make check

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
passed [qa](https://pulpito.ceph.com/cbodley-2022-08-24_22:54:44-rgw-wip-cbodley-testing-distro-default-smithi/) with [rerun](https://pulpito.ceph.com/cbodley-2022-08-25_13:25:55-rgw-wip-cbodley-testing-distro-default-smithi/)

**Pull request merged**